share/spice/currency/currency.js

```diff
@@ -286,14 +278,17 @@
                 $left_select.val(from);
             }
 
-            var endpoint = "/js/spice/currency/1/" + from + "/" + to;
-            $.get(endpoint, function(payload) {
-
-                // jsonp is returned from the API so we have to alter the contents
+            $.ajax({
+                url: "/js/spice/currency/1/" + from + "/" + to,
+                beforeSend: function( xhr ) {
+                    xhr.overrideMimeType( "text/plain; charset=x-user-defined" );
+                },
+                error: function(request, status, error) {
+                    Spice.failed('currency');
+                }
+            }).done(function(payload) {
                 var response = JSON.parse(payload.trim().replace(/^[^\(]*\(/, '').replace(/\);$/, ''));
```

Comment:
Perhaps we should just make an `alt_to` in the Perl so we don't need to parse the response via regex -- that seems like a much safer solution.